server/src/main/java/org/elasticsearch/cluster/service/MasterService.java

```diff
@@ -569,7 +569,7 @@ public TimeValue ackTimeout() {
 
     protected void warnAboutSlowTaskIfNeeded(TimeValue executionTime, String source) {
         if (executionTime.getMillis() > slowTaskLoggingThreshold.getMillis()) {
-            logger.warn("cluster state update task [{}] took [{}] which is above the warn threshold of {}", source, executionTime,
+            logger.warn("cluster state update task [{}] took [{}] which is above the warn threshold of [{}]", source, executionTime,
```

Comment:
An interesting follow-up would be to change the slow logging in MasterService so that it only applies to `calculateTaskOutputs`, and then reduce the timeout from 30s to 10s. Currently,  the slow logging in MasterService measures cluster state update calculation, publication +  application. The latter two are covered by your PR here now

Reply:
++ good idea